result.h

```diff
@@ -844,6 +844,20 @@ struct Result {
         std::terminate();
     }
 
+    template<typename F = E>
+    typename std::enable_if<
+        !std::is_same<F, void>::value,
+        F
+    >::type
```

Comment:
Why isn't a void error type allowed? I can see that not being too helpful in production code, but having a void error type could still be useful for prototyping or simple applications. It's also allowed in Rust (as `Result<T, ()>`).